src/utils/customHTML.ts

```diff
@@ -0,0 +1,109 @@
+}
+
+/**
+ * Retrieves the custom HTML tag for embedding an Instagram post
+ * @param postId the ID of the Instagram post
+ * @param isCaptioned whether to include the post caption
+ * @returns the Instagram post custom HTML tag code
+ */
+export const getInstagramEmbedTag = (
+  postId: string | undefined,
+  isCaptioned: boolean
+): string => {
+  const extraAttr = isCaptioned ? " captioned" : ""
+
+  if (!postId) {
+    return ""
+  }
+
+  return `<instagram-post post-id="${postId}" ${extraAttr}></instagram-post>`
+}
+
+/**
+ * Processes the given HTML chunk by converting the custom Instagram tag
+ * into the appropriate HTML for embedding the Instagram post.
+ * Refer to getInstagramEmbedTag() for the custom tag format.
+ * @param html the HTML chunk to process
+ * @returns the processed HTML chunk
+ */
+export const processInstagramTagToEmbed = (html: string): string => {
+  const $ = cheerio.load(html, null, false)
+  $("instagram-post").each((i: number, el: Element) => {
+    const postId = $(el).attr("post-id")
+    const isCaptioned = !!(
+      ($(el).attr("captioned") || $(el).attr("captioned") === "") &&
+      $(el).attr("captioned") !== "false"
```

Comment:
this is pretty confusing to untangle ._.

wdyt about writing it as follows: 
```suggestion
      $(el).attr("captioned") && $(el).attr("captioned") !== "false" || $(el).attr("captioned") === "" 
```
which implies that the `captioned` attribute exists and is not "false" or that it is an empty caption

Reply:
Thanks for this suggested change! I was having quite some difficulty trying to simplify this logic.